src/FindIntrinsics.cpp

```diff
@@ -623,12 +623,16 @@ Expr lower_rounding_shift_right(const Expr &a, const Expr &b) {
 
 Expr lower_saturating_add(const Expr &a, const Expr &b) {
     internal_assert(a.type() == b.type());
-    return saturating_narrow(widening_add(a, b));
+    // Lower saturating add without using widening arithmetic, which may require
+    // types that aren't supported.
+    return simplify(clamp(a, a.type().min() - min(b, 0), a.type().max() - max(b, 0))) + b;
 }
 
 Expr lower_saturating_sub(const Expr &a, const Expr &b) {
     internal_assert(a.type() == b.type());
-    return saturating_cast(a.type(), widening_sub(a, b));
+    // Lower saturating add without using widening arithmetic, which may require
+    // types that aren't supported.
+    return simplify(clamp(a, a.type().min() - max(b, 0), a.type().max() - min(b, 0))) - b;
```

Comment:
Hmm, looking at this, I'm not sure how it works. It seems like these limits should overflow. But the test should be solid here...

Reply:
I think this is some accidental and magical 2's complement overflow trick... anyways fixed.